packages/auth/lib/challenges/SMSCodeChallenge.js

```diff
@@ -0,0 +1,89 @@
+const {
+  sendTextMessage
+} = require('@orbiting/backend-modules-sms')
+const { parse, format } = require('libphonenumber-js')
+
+const MIN_IN_MS = 1000 * 60
+
+function generateSMSTokenCode () {
+  // 2’176’782’336 possibilities, assuming 30 ms takes
+  // 700 days to bruteforce the code through the API
+  let text = ''
+  const possible = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789'
+  for (var i = 0; i < 6; i++) {
+    text += possible.charAt(Math.floor(Math.random() * possible.length))
+  }
+  return text
+}
+
+async function getUserPhoneNumber (pgdb, email) {
+  const user = await pgdb.public.users.findOne({ email })
+  try {
+    const parsedPhoneNumber = parse(user.phoneNumber || '', 'CH') // it could be any arbitrary string
+    return format(parsedPhoneNumber.phone, parsedPhoneNumber.country, 'E.164')
+  } catch (e) {
+    throw new Error('Phone number not valid for Text Messaging')
+  }
+}
+
+module.exports = {
+  generateSharedSecret: async ({ pgdb, user, email }) => {
```

Comment:
where does email come from? as far as i can see [resendPhoneNumberVerificationCode.js](packages/auth/graphql/resolvers/_mutations/resendPhoneNumberVerificationCode.js) only provides user, not email.

Reply:
fixed